src/Components/ScrivitoExtensions/ChatbotTab/useChatCompletion.js

```diff
@@ -0,0 +1,117 @@
+import OpenAI from "openai";
+import { useMemo, useState } from "react";
+
+export function useChatCompletion({ apiKey, instanceId, model, user }) {
+  const [messages, setMessages] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [completionMessage, setCompletionMessage] = useState(null);
+
+  const messagesWithCompletion = useMemo(
+    () => (completionMessage ? messages.concat(completionMessage) : messages),
+    [messages, completionMessage]
+  );
+
+  return {
+    loading,
+    messages: messagesWithCompletion,
+    submitPrompt: (prompt) => {
+      setCompletionMessage(null);
+      setLoading(true);
+      // @ts-ignore
+      setMessages((m) => {
+        const messagesWithPrompt = [...m, ...prompt];
+        startStreaming({
+          apiKey,
+          instanceId,
+          model,
+          messages: messagesWithPrompt,
+          setCompletionMessage,
+          setLoading,
+          setMessages,
+          user,
+        });
+        return messagesWithPrompt;
+      });
+    },
+    abortResponse: () => {},
+    resetMessages: () => {
+      setCompletionMessage(null);
+      setMessages([]);
+    },
+    setMessages,
+  };
+}
+
+let OPENAI_API_KEY =
+  // @ts-ignore
+  typeof import.meta.env === "undefined"
+    ? // @ts-ignore
+      process.env.OPENAI_API_KEY
+    : // @ts-ignore
+      import.meta.env.OPENAI_API_KEY;
+
+async function startStreaming({
+  apiKey,
+  instanceId,
+  messages,
+  model,
+  setCompletionMessage,
+  setLoading,
+  setMessages,
+  user,
+}) {
+  const client = new OpenAI({
+    apiKey: OPENAI_API_KEY || apiKey,
+    baseURL: OPENAI_API_KEY
+      ? "https://api.openai.com/v1"
+      : "https://i7ukqy3mhy3nzkn3dutmmzdx440xgtjk.lambda-url.eu-west-1.on.aws?ignore=",
+
+    defaultQuery: { tenant_id: instanceId },
+    defaultHeaders: { Accept: "*/*" },
+    dangerouslyAllowBrowser: true,
+    fetch: async (url, init) => {
+      return fetch(url, {
+        ...init,
+        headers: cleanHeaders(init?.headers),
+        body: JSON.stringify(JSON.parse(init?.body)),
+      });
+    },
+  });
+
+  const stream = await client.beta.chat.completions.stream({
+    model,
+    messages,
+    stream: true,
+    user,
+  });
+
+  stream.on("error", () => {
+    setLoading(false);
+  });
+
+  stream.on("content", () => {
+    const message = stream.currentChatCompletionSnapshot?.choices[0].message;
+    if (message) setCompletionMessage({ ...message });
+  });
+
+  return stream.finalChatCompletion().then(({ choices }) => {
+    setCompletionMessage(null);
+    const { tool_calls, ...otherMessageAttributes } = choices[0].message;
```

Comment:
@markus-codewrangler You write that `tool_calls` causes an error 400, that's why you need to remove it here.

But isn't that a problem caused by our own backend? I yes, then it would be better to fix the backend, IMHO.

Reply:
I was under the impression that our AI-Proxy just pipes the response unchanged. Did I overlook something? 🤔 

I suspect this might be a bug in the OpenAI SDK's beta API:

https://github.com/dcsaszar/scrivito-sam/blob/e74e12f740f4a72c870a5a35604343613b578a11/src/Components/ScrivitoExtensions/ChatbotTab/useChatCompletion.js#L78

Reply:
Before assuming that this is OpenAI's bug, we should first analyze what's going on and verify where exactly the error comes from and whether it really isn't our own fault.

Reply:
BTW FMI What's the reason to use the "beta"-API here? 
Which advantage does it have over the regular API?

Reply:
I'm asking because I switched back to the regular API in my own code - seems easier to work with for me.

Reply:
We could probably use the regular API. We'd just have to adapt the hook a bit.

Reply:
From the repo it looks like beta is likely to have issues. At the time I just assumed beta is recommended b/c the readme has a lot of beta examples.